OptimizelySDKCore/OptimizelySDKCore/NSString+NSString_OPTLY.h

```diff
@@ -0,0 +1,13 @@
+//
+//  NSString+NSString_OPTLY.h
+//  OptimizelySDKCore
+//
+//  Created by Abdur Rafay on 08/01/2018.
+//  Copyright © 2018 Optimizely. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@interface NSString (NSString_OPTLY)
+- (BOOL)isEmpty;
+@end
```

Comment:
In my previous comment regarding OPTLYMacros.h I wrote:

    I'd like to suggest adding a "stringOrEmpty:" method to OPTLYEventBuilder.m and
    "isEmptyString:" and "isEmptyArray:" methods to OPTLYDecisionService.m . My
    comment can be taken as asking you to code these 3 methods, use these 3 methods
    instead of the macros, and delete OPTLYMacros.h from the workspace. These 3 methods
    could be "static" "+" methods in those 2 classes.

ISSUES: Instead, you created an NSString category.  This has problems.
* When Optimizely code dynamically loads the category, it adds
a method "isEmpty" to Apple's NSString .  This will cause a name
collision with an "isEmpty" which can be defined in customer's own
NSString category.
* The public NSString category approach is less encapsulated
than private OPTLYEventBuilder class method approach which I've
recommended.
* The actual implementation of the "- (BOOL)isEmpty" turns
out to be doing something a bit different from the
    #define isEmptyString(A)  ({ !(A && [A respondsToSelector:@selector(isEqualToString:)] && ![(NSString *)A isEqualToString:@""]); })
you created in your earlier proposed commit change to OPTLYMacros.h .
Passing over this difference, my impression is
    [self isEqualToString:@""]
is too simple to warrant creation of a method and around that
a category.

THEREFORE: I'd like to ask you to not create NSString+NSString_OPTLY.m+h .
I'd prefer one of these approaches:
* If comparison of string with @"" in OPTLYEventBuilder.m
code is indeed sufficient for you, then just code
    [s isEqualToString:@""]
in OPTLYEventBuilder.m where you need this comparison.
* OTOH, if you do need semantics more like the "#define isEmptyString(A) ..."
macro you had, then you could code
    + (BOOL)isEmptyString:(NSObject*)s {
        return (!s
                || ![s isKindOfClass:[NSString class]]
                || [(NSString *)s isEqualToString:@""]);
    }